src/CompletionHandler.php

```diff
@@ -257,10 +267,18 @@ protected function completeForCommandName()
     {
         if (!$this->command || (count($this->context->getWords()) == 2 && $this->context->getWordIndex() == 1)) {
             $commands = $this->application->all();
-            $names = array_keys($commands);
+            $names = array();
```

Comment:
The `$names` array is the same, that `$helpCommands` array (if you make `_completion` command hidden). I recommend extracting logic for getting non-hidden commands to separate methods and calling in both places.